collectors/ebpf.plugin/ebpf.c

```diff
@@ -633,6 +638,557 @@ static inline int parse_disable_apps(char *ptr)
+    link_hostnames(&network_viewer_opt.included_hostnames, value);
+
+    value = appconfig_get(&collector_config, EBPF_NETWORK_VIEWER_SECTION, "excluded hostnames", NULL);
+    link_hostnames(&network_viewer_opt.excluded_hostnames, value);
+
+    value = appconfig_get(&collector_config, EBPF_NETWORK_VIEWER_SECTION, "included ips", NULL);
+    parse_values((void **)&network_viewer_opt.included_ips, value, isascii, parse_ip_list);
+
+    value = appconfig_get(&collector_config, EBPF_NETWORK_VIEWER_SECTION, "excluded ips", NULL);
+    parse_values((void **)&network_viewer_opt.excluded_ips, value, isascii, parse_ip_list);
+}
+
+/**
+ * Link dimension name
+ *
+ * Link user specified names inside a link list.
+ *
+ * @param port the port number associated to the dimension name.
+ * @param hash the calculated hash for the dimension name.
+ * @param name the dimension name.
+ */
+static void link_dimension_name(char *port, uint32_t hash, char *value)
+{
+    int test = str2i(port);
+    if (test < NETDATA_MINIMUM_PORT_VALUE || test > NETDATA_MAXIMUM_PORT_VALUE){
+        error("The dimension given (%s = %s) has an invalid value and it will be ignored.", port, value);
+        return;
+    }
+
+    ebpf_network_viewer_dim_name_t *w;
+    w = callocz(1, sizeof(ebpf_network_viewer_dim_name_t));
+
+    w->name = strdupz(value);
+    w->hash = hash;
+
+    w->port = (uint16_t) test;
+
+    ebpf_network_viewer_dim_name_t *names = network_viewer_opt.names;
+    if (unlikely(!names)) {
+        network_viewer_opt.names = w;
+    } else {
+        for (; names->next; names = names->next) {
+            if (names->port == w->port) {
+                error("Dupplicated definition for the service '%u', the name %s will be ignored. ", hash, names->name);
+                freez(names->name);
+                names->name = w->name;
+                names->hash = w->hash;
+                freez(w);
+                return;
+            }
+        }
+        names->next = w;
+    }
+
+#ifdef NETDATA_INTERNAL_CHECKS
+    info("Adding values %s( %u) to dimension name list used on network viewer",w->name, w->port);
+#endif
+}
+
+/**
+ * Parse service Name section.
+ *
+ * This function gets the values that will be used to overwrite dimensions.
+ */
+static void parse_service_name_section()
+{
+    struct section *co = appconfig_get_section(&collector_config, EBPF_SERVICE_NAME_SECTION);
+    if (co) {
+        struct config_option *cv;
+        for (cv = co->values; cv ; cv = cv->next) {
+            link_dimension_name(cv->name, cv->hash, cv->value);
+        }
+    }
+
+    //Always associated the default port to Netdata
+    ebpf_network_viewer_dim_name_t *names = network_viewer_opt.names;
+    uint16_t default_port = htons(19999);
+    for (; names->next; names = names->next) {
+        if (names->port == default_port) {
+            return;
+        }
+    }
+
+    link_dimension_name("Netdata", simple_hash("Netdata"), "19999");
```

Comment:
in line `1176` you call `link_dimension_name(PORT, HASH, NAME);` but
in line `1189` you call `link_dimension_name(NAME, HASH, PORT);` ?

also is it good idea to have fixed 19999 ? what if user changes the port netdata listens on?

Reply:
`cv->name` here holds key name which in config as you described in PR description (and how I tested it) looks like this:

```
[service name]
    2222 = testing
```

then `cv->name` is the `port`. So I guess either configuration description is wrong or code. Or I misunderstand something?

Reply:
You are right, in the second call (line 1189), I inverted, because in normal situation I do not use numbers as name.

Reply:
>   //Always associated the default port to Netdata

What purpose does it serve? Why we do it? 🤔